src/JSInterop/Microsoft.JSInterop/src/JSRuntimeBase.cs

```diff
@@ -62,18 +66,35 @@ public Task<T> InvokeAsync<T>(string identifier, params object[] args)
         /// <param name="argsJson">A JSON representation of the arguments.</param>
         protected abstract void BeginInvokeJS(long asyncHandle, string identifier, string argsJson);
 
+        /// <summary>
+        /// Allows derived classes to configure the information about an exception in a JS interop call that gets sent to JavaScript.
+        /// </summary>
+        /// <remarks>
+        /// This callback can be used in remote JS interop scenarios to sanitize exceptions that happen on the server to avoid disclosing
+        /// sensitive information to remote browser clients.
+        /// </remarks>
+        /// <param name="exception">The exception that occurred.</param>
+        /// <param name="assemblyName">The assembly for the invoked .NET method.</param>
+        /// <param name="methodIdentifier">The identifier for the invoked .NET method.</param>
+        /// <returns>An object containing information about the exception.</returns>
+        protected virtual object OnDotNetInvocationException(Exception exception, string assemblyName, string methodIdentifier) => exception;
```

Comment:
Remote JS runtime will override this and sanitize exceptions before serializing them.

Reply:
Since the TS-side code expects the payload to be a `string` whenever `success == false`, would it be safer to declare the return type of this method to be `string`? Is there any scenario where we want it to be something else?

Reply:
Whatever it is, will end up being serialized into a string. I don't want to limit ourselves in the future from returning an object here with a bit more information.

For example, the idea is we will return an error message crafted from the server, but we might as well return an object with Assembly, Method, Description and have the client build the actual error.

In the end the client receives a string that might contain json in it, does that make sense?

Reply:
Ok, makes sense!